zerver/tests/test_signup.py

```diff
@@ -274,7 +274,7 @@ def test_register(self):
         with queries_captured() as queries:
             self.register(self.nonreg_email('test'), "test")
         # Ensure the number of queries we make is not O(streams)
-        self.assert_length(queries, 66)
+        self.assert_length(queries, 106)
```

Comment:
OK, looks great!  I briefly merged this commit and then backed it out.  I backed it out because we may want to do a backfill migration similar to what we have in zerver/migrations/0057_realmauditlog.py.  @rishig what are your thoughts on whether we want some sort of backfill here?  We can't of course go back to the beginning of time, but we could do `subscription_created` (and `subscription_deactivated` entries) with `backfilled=True` based on the current state as of this commit, and I think that'd be a good idea for making it easy to assume things in code like "there is a `subscription_created` RealmAuditLog entry for every actual subscription to a stream"..  

As a sidenote, if you have some spare time for doing relatively low-complexity work, replacing the rest of `log_event()`  calls with `RealmAuditLog` entries is a goal of ours.  But that belongs in a separate PR.  In general, I don't think we migrations to create pre-existing state, but for this field in particular, it seems like a good idea.